mediator/med_fraction_mod.F90

```diff
@@ -299,7 +300,23 @@ subroutine med_fraction_init(gcomp, rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
        call fldbun_getdata1d(is_local%wrap%FBFrac(compocn), 'ofrac', ofrac, rc)
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       ofrac(:) = So_omask(:)
+       if (associated(So_omask)) then
```

Comment:
I don't quite understand why this is required. Won't the call to fldbun_getdata1d return an error if So_omask doesn't exist in FBImp(compocn,compocn)? 

Why isn't a similar change required at LN 249 (setting Si_imask and ifrac)?

Reply:
Hi @DeniseWorthen . It doesn't return an error and you don't want it to return an error. I found this because the ocean decomposition is sparse, meaning there are PETs that don't allocate any ocean cells. In this case FBFrac(compocn) returns a null pointer. It can be fixed another way through allocating zero sized objects but that fix has to go deeper into the code.

A similar change should be made for ifrac but HAFS doesn't have an ice component and therefore it didn't crash. I can replicate this code for the other fractions but I won't be able to test the change.

Reply:
This work is proving me a little more trouble. In some cases the logic is complex. Would you approve a change that looked more like this? In this case it simplifies the check but takes care of my unique case.
```
       if (associated(ofrac)) then
          ofrac(:) = So_omask(:)
       endi
```